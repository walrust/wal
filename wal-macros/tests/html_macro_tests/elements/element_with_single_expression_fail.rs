use wal_macros::html;

include!("../utils/non_display_struct.rs");

fn main() {
    html! { <div>()</div> };
    html! { <div>{ }</div> };
    html! { <div> { invalid_reference } </div> };
    html! { <div> String::from("Hello world!") </div> };
    let val = "Hello world!";
    html! { <div> val </div> };
    let t = NonDisplayStruct;
    html! { <div> { t } </div> };
    html! { <div> { NonDisplayStruct } </div> };
    let node = || NonDisplayStruct;
    html! { <div> { node() } </div> };
}
