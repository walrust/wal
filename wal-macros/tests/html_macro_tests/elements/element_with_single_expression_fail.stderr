error: Invalid syntax encountered
 --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:8:18
  |
8 |     html! { <div>()</div> };
  |                  ^

error: Expected expressions within braces, found an empty block
 --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:9:18
  |
9 |     html! { <div>{ }</div> };
  |                  ^^^

error: Invalid syntax encountered
  --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:11:19
   |
11 |     html! { <div> String::from("Hello world!") </div> };
   |                   ^^^^^^

error: Invalid syntax encountered
  --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:13:19
   |
13 |     html! { <div> val </div> };
   |                   ^^^

error[E0425]: cannot find value `invalid_reference` in this scope
  --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:10:21
   |
10 |     html! { <div> { invalid_reference } </div> };
   |                     ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0277]: `TestNonDisplayStruct` doesn't implement `std::fmt::Display`
  --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:15:19
   |
15 |     html! { <div> { t } </div> };
   |                   ^^-^^
   |                   | |
   |                   | this tail expression is of type `TestNonDisplayStruct`
   |                   `TestNonDisplayStruct` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `TestNonDisplayStruct`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = help: the following other types implement trait `From<T>`:
             <VNode as From<T>>
             <VNode as From<VElement>>
             <VNode as From<VList>>
             <VNode as From<VText>>
   = note: required for `TestNonDisplayStruct` to implement `ToString`
   = note: required for `VNode` to implement `From<TestNonDisplayStruct>`

error[E0277]: `TestNonDisplayStruct` doesn't implement `std::fmt::Display`
  --> tests/html_macro_tests/elements/element_with_single_expression_fail.rs:16:19
   |
16 |     html! { <div> { TestNonDisplayStruct { field: 15 } } </div> };
   |                   ^^----------------------------------^^
   |                   | |
   |                   | this tail expression is of type `TestNonDisplayStruct`
   |                   `TestNonDisplayStruct` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `TestNonDisplayStruct`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = help: the following other types implement trait `From<T>`:
             <VNode as From<T>>
             <VNode as From<VElement>>
             <VNode as From<VList>>
             <VNode as From<VText>>
   = note: required for `TestNonDisplayStruct` to implement `ToString`
   = note: required for `VNode` to implement `From<TestNonDisplayStruct>`
